go/pkg/pwapi/api_challenge-subscription-validate.go

```diff
@@ -53,7 +53,16 @@ func (svc *service) ChallengeSubscriptionValidate(ctx context.Context, in *Chall
 	// compare input and instances' passphrases
 	var amountExpected int
 	{
-		configData, err := instances[0].ParseInstanceConfig()
+		var availableInstance *pwdb.ChallengeInstance
+		for _, instance := range instances {
+			if instance.Status == pwdb.ChallengeInstance_Available {
+				availableInstance = instance
+			}
+		}
+		if availableInstance == nil {
+			return nil, errcode.ErrNoAvailableChallengeInstance
+		}
+		configData, err := availableInstance.ParseInstanceConfig()
```

Comment:
this part however will only take the last available instance, since we can have multiple available instances at the same time

Reply:
I think that the script was already correct for the case when we have at least one available instance

So I suggest you to just make a `found := false` check here

Reply:
oh OK I expanded the github diff preview, and finally that's ok :+1: